### PR TITLE
[BugFix] fix generating endlessly bug when reasoner grammar & spec decoding are enabled

### PR DESCRIPTION
**Pull request by justadogistaken**
## Motivation

Related to #9187 and #11550.

When reasoner grammar and spec decoding are enabled at the same time.
The model will generate `<think>` first, and nothing would be generated till reach `max_tokens`.

I found that the problem comes from `allocate_token_bitmask.fill_(0)` and `reasoner grammar`.

```
def traverse_tree(
    retrieve_next_token: torch.Tensor,
    retrieve_next_sibling: torch.Tensor,
    draft_tokens: torch.Tensor,
    grammar: BaseGrammarObject,
    allocate_token_bitmask: torch.Tensor,
):
    """
    Traverse the tree constructed by the draft model to generate the logits mask.
    """
    assert (
        retrieve_next_token.shape == retrieve_next_sibling.shape == draft_tokens.shape
    )

    allocate_token_bitmask.fill_(0)

    def dfs(
        curr: int,
        retrieve_next_token: torch.Tensor,
        retrieve_next_sibling: torch.Tensor,
        parent_pos: int,
    ):
```

## Checklist

- [ ] Format your code according to the [Format code with pre-commit](https://docs.sglang.ai/developer_guide/contribution_guide.html#format-code-with-pre-commit).
- [ ] Add unit tests according to the [Run and add unit tests](https://docs.sglang.ai/developer_guide/contribution_guide.html#run-and-add-unit-tests).
- [ ] Update documentation according to [Write documentations](https://docs.sglang.ai/developer_guide/contribution_guide.html#write-documentations).
- [ ] Provide accuracy and speed benchmark results according to [Test the accuracy](https://docs.sglang.ai/developer_guide/contribution_guide.html#test-the-accuracy) and [Benchmark the speed](https://docs.sglang.ai/developer_guide/contribution_guide.html#benchmark-the-speed).

### CODE REVIEW

**Comment by justadogistaken** [comment]
# Test
Problem Solved with this pr. And this has been deployed in production.

# Request

```
curl 'http://localhost:30000/v1/chat/completions' \
-H 'Content-Type: application/json' \
-d '{
    "model": "Qwen/Qwen3-8B", 
    "messages": [ 
        {
            "role": "system",
            "content": "You are a highly advanced and versatile AI assistant."
        },
        {
            "role": "user",
            "content": "Tell me about the latest advancements in AI."
        }
    ],
    "max_tokens": 1000,
    "response_format": {
        "type": "json_object"
    }
}'
```

image

image


image

**Comment by justadogistaken** [comment]
@LorrinWWW The problem seems to be solved in the latest version. I will close the pr. But I found there is another problem. Could you help review this?

ref #14480
